proposals/000-decorator-api.md

```diff
@@ -0,0 +1,322 @@
+    * You should check whether the Navigator supports drawing the decoration styles required by a particular feature before enabling it. For example, underlining an audiobook does not make sense, so an Audiobook Navigator would not support the `underline` decoration style.
+* `supportsDecorationInteractionForStyle(interaction: Decoration.Interaction, style: Decoration.Style) -> Boolean`
+    * Indicates whether the Navigator supports the given `interaction` for this `style`.
+    * You should check whether the Navigator supports the interactions required by a particular feature before enabling it. Alternatively, you could offer a fallback in the user interface.
+
+#### Events
+
+* `DecorationActivated(Decoration)`
+    * Emitted when the user activates a decoration. For example, when clicking on a highlight.
+
+### `Decoration` Class
+
+A decoration is a user interface element drawn on top of a publication. It associates a style to be rendered with a precise location (`Locator`) in the publication.
+
+For example, decorations can be used to draw highlights, images or buttons.
+
+#### Properties
+
+* `id: String`
+    * An identifier for this decoration. It must be unique in the group the decoration is applied to.
+* `locator: Locator`
+    * Location in the publication where the decoration will be rendered.
+* `style: Decoration.Style`
+    * Declares the look and feel of the decoration.
+* `interaction: Set<Decoration.Interaction>`
+    * List of enabled user interactions for this decoration.
+
+### `Decoration.Interaction` Enum
+
+A type of user interaction with a decoration.
+
+* `Activate`
+    * Primary interaction with a decoration, such as a single tap or click.
+
+The user can interact with a decoration if it is enabled in `decoration.interactions` and supported by the Navigator for its style.
+
+The terminology avoids concrete gestures (e.g. tap) to fit all platforms (mouse vs. touch) and media type. For example, activating a decoration for an Audiobook Navigator might be very different from activating a visual highlight.
+
+### `Decoration.Style` Interface
+
+The Decoration Style determines the look and feel of a decoration once rendered by a Navigator. It is media type agnostic, meaning that each Navigator will translate the style into a set of rendering instructions which makes sense for the resource type.
+
+The Readium toolkit supports two default styles:
+
+* `Highlight(tint: Color?)`
+* `Underline(tint: Color?)`
+
+**Note**: This can be implemented differently depending on the platform capabilities. Ideally, this is a marker interface and each concrete type is used to identify the style.
+
+### Decoration Templates
+
+Decoration Templates translate a `Decoration` object and its associated style into concrete rendering instructions specific to a Navigator and resource type. Each concrete implementation of `DecorableNavigator` can support its own kind of decoration templates.
```

Comment:
If we opt for a unique generic navigator, I think we must find a way to associate templates with media types. Maybe associating specific view classes to media types, kind of as we do in the Streamer API. Maybe the whole proposal should be revisited later within the new navigator architecture framework.

Reply:
Agreed